core/src/main/java/org/jdbi/v3/core/Handle.java

```diff
@@ -122,47 +124,59 @@ public Handle setStatementBuilder(StatementBuilder builder) {
      */
     @Override
     public void close() {
+        final List<Throwable> suppressed = new ArrayList<>();
         if (closed) {
             return;
         }
 
-        boolean wasInTransaction = isInTransaction()
-            && forceEndTransactions
-            && config.get().get(Handles.class).isForceEndTransactions();
+        boolean wasInTransaction = forceEndTransactions
+                && config.get().get(Handles.class).isForceEndTransactions();
+
+        if (wasInTransaction) {
+            try {
+                wasInTransaction &= isInTransaction();
```

Comment:
Can we move this stuff into `isInTransaction()` instead? We could probably leave this method as is if `isInTransaction()` checked whether the connection was already closed--unless I'm missing something.

Reply:
I don't think so.  If we move the exception catching into `isInTransaction` we are then forced to handle the exception then and there and invent a fallback `boolean` value -- do we pretend to be in a transaction?  or not?  neither is right all the time.  so I don't think the contract of `isInTransaction` should change -- if it can't determine it, it should throw.

Reply:
This guards against arbitrary problems -- imagine an `EOFException` while trying to query the server's transaction status -- not just concurrent closes.